packages/remix-dev/vite/plugin.ts

```diff
@@ -1131,7 +1150,7 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
                   nodeReq,
                   nodeRes
                 ) => {
-                  let req = fromNodeRequest(nodeReq);
+                  let req = fromNodeRequest(nodeReq, build.publicPath !== "/");
```

Comment:
use the `originalUrl` when `publicPath` is not `/` indicating that a `base` was set and vite will alter `req.url`